### DISCUSSION TITLE
List does not returns ReqIF defintion

### DISCUSSION THREAD

**Discussion by reqTibi** [IBM/ELM-Python-Client#37] [Q&A]
Hello,

i try to get the ReqIF definition from our prod server but unfortunately the result is 0 for all tested components.
call:
`reqif_io.exe -C "Non_Functional_Requirements" -F "Non_Functional_Requirements_COP-S" "Control Unit System Development (Requirements)" list`

the last query:
https://JAZZ_URL/rm/reqif_oslc/definitions/query?componentURI=https%3A//JAZZ_URL/rm/cm/component/_H3JG42_KEeiaocRoWLkjzw&oslc.pageSize=200&oslc.paging=true&oslc.select=%2A

returns:
`<rdf:RDF>
<oslc:ResponseInfo rdf:about="https://JAZZ_URL/rm/reqif_oslc/definitions/query">
<dcterms:title>Query Results : 0</dcterms:title>
</oslc:ResponseInfo>
</rdf:RDF>`

but there is a definition.
The component and project idenfier inside the request (from log file) seems to be correct.

On our test server (identical setup, same version + ifix, same user + password from same AD) it work as expected.

Can you give me a hint, what could be the problem?

**Comment by reqTibi** [comment]
One point, which differs between the systems, but I do not think is relevant for this problem. On the test system the user is not forced to use change sets on the productive system this feature is activated.

**Comment by barny** [comment]
Once I've manually created a reqif definition (in a cs if cs are required, then delivered it) then using `list` shows the definition even when cs are required.

But it appears that when cs are required the `export` requires a changeset, so I get a 401 Forbidden, this is presumably because export updates the component with the reqif and the export report.

Ah there's a typo in reqif_io.py; change this line (around line 233):
    if app.supports_components:
to:
    if mainapp.supports_components:

AFAICT that typo would prevent selecting any non-default configuration.

If you make that change you should be able to use the -F flag to specify an already-existing changeset by name to do the export, can you confirm? If so I'll deliver this change in a few days.

I don't have any plans to implement the creation of a changeset in the example reqif_io.py, and I don't think this is implemented in elmclient. I may extend elmclient with changeset handling and write examples of create/deliver/discard changeset, but that won't be soon.

You are of course very welcome to clone the repository and modify it then put in a pull request so I can consider adopting your edits :-)

**Comment by barny** [comment]
Release 0.17.0 fixes the configuration selection problem

**Reply by reqTibi** [reply]
Thank you very much. I tried to address a change set via parameter -F as Name and also as URL. How can address a Change set for the export?

The new version works fine for definition list output.
